multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/OperationsApiServiceImpl.java

```diff
@@ -214,26 +209,13 @@ private Operation getOperation(String operationId) {
             Operation operation = operationService.createQuery()
                                                   .processId(operationId)
                                                   .singleResult();
-            operation = addOperationState(operation);
             operation = operationsHelper.addErrorType(operation);
-            return operation;
+            return operationsHelper.releaseLockIfNeeded(operation);
```

Comment:
Will this lead to more db queries when we get operation?

Reply:
Will lead to 1 more db query only in case the lock is not released and the operation is in final state. Apparently there was/is an inconsistency in flowable tables, which this method fixes.